include/SFML/Uchar.h

```diff
@@ -22,22 +22,21 @@
 //
 ////////////////////////////////////////////////////////////
 
-#ifndef SFML_MUTEXSTRUCT_H
-#define SFML_MUTEXSTRUCT_H
-
-////////////////////////////////////////////////////////////
-// Headers
-////////////////////////////////////////////////////////////
-#include <SFML/System/Mutex.hpp>
+#ifndef SFML_UCHAR_H
+#define SFML_UCHAR_H
 
 
 ////////////////////////////////////////////////////////////
-// Internal structure of sfMutex
+// Define a macro for char32_t for MacOS
 ////////////////////////////////////////////////////////////
-struct sfMutex
-{
-    sf::Mutex This;
-};
+#include <SFML/Config.h>
 
+#ifndef CSFML_SYSTEM_MACOS
+    #include <uchar.h>
+    typedef char32_t sfChar32;
+#else
+    #include <stdint.h>
+    typedef uint_least32_t sfChar32;
+#endif
```

Comment:
Let's put this stuff in Config.h. That's where the old typedefs used to live before I recently removed them.

Reply:
Moved to config, let me know if any more changes are needed.